File: apps/backend/internal/apprelease/model/app_release.go
package model

import "time"

const (
	PlatformAndroid = "android"
	PlatformIOS     = "ios"
	PlatformWindows = "windows"
	PlatformMacOS   = "macos"
	PlatformLinux   = "linux"

	StatusDraft     = "draft"
	StatusPublished = "published"
	StatusDisabled  = "disabled"
)

type AppRelease struct {
	ID                  uint   `gorm:"primaryKey"`
	Platform            string `gorm:"size:32;not null"`
	VersionName         string `gorm:"size:64;not null"`
	VersionCode         int64  `gorm:"not null"`
	Title               string `gorm:"size:255;not null"`
	Changelog           string `gorm:"not null"`
	DownloadURL         string `gorm:"size:2048;not null"`
	FileSize            *int64
	FileSHA256          *string `gorm:"size:64"`
	MinimumVersionCode  int64   `gorm:"not null"`
	ForceUpdate         bool    `gorm:"not null"`
	Status              string  `gorm:"size:32;not null"`
	PublishedAt         *time.Time
	AnnouncementID      *uint
	AnnouncementManaged bool
	CreatedAt           time.Time
	UpdatedAt           time.Time
}

File: apps/frontend/app/types/appRelease.ts
export type AppReleasePlatform =
  | 'android'
  | 'ios'
  | 'windows'
  | 'macos'
  | 'linux'

export type AppReleaseStatus = 'draft' | 'published' | 'disabled'

export interface AppRelease {
  id: number
  platform: AppReleasePlatform
  versionName: string
  versionCode: number
  title: string
  changelog: string
  downloadUrl: string
  fileSize?: number | null
  sha256?: string | null
  minimumVersionCode: number
  forceUpdate: boolean
  status: AppReleaseStatus
  publishedAt?: string | null
  announcementId?: number | null
  createdAt: string
  updatedAt: string
}

export interface AppReleasePayload {
  platform: AppReleasePlatform
  versionName: string
  versionCode: number
  title: string
  changelog: string
  downloadUrl: string
  fileSize?: number | null
  sha256?: string | null
  minimumVersionCode: number
  forceUpdate: boolean
  createAnnouncement: boolean
  status: AppReleaseStatus
  publishedAt?: string | null
  announcementId?: number | null
}

export interface GitHubReleaseAsset {
  name: string
  size: number
  sha256?: string | null
  downloadUrl: string
  updatedAt: string
}

export interface GitHubRelease {
  tag: string
  name: string
  body: string
  prerelease: boolean
  createdAt: string
  publishedAt: string
  apk?: GitHubReleaseAsset | null
}

export interface LatestAppRelease {
  hasUpdate: boolean
  forceUpdate?: boolean
  latestVersion?: { versionName: string; versionCode: number }
  title?: string
  changelog?: string
  downloadUrl?: string
  fileSize?: number | null
  sha256?: string | null
  publishedAt?: string
}
